diff --git a/dist/esm/rpc/index.js b/dist/esm/rpc/index.js
index 53b573491bf5fef4022b3e6585e1ba019c9749ab..48b2c77c42833c78ed08d31ef9cd9ea9d6be70ea 100644
--- a/dist/esm/rpc/index.js
+++ b/dist/esm/rpc/index.js
@@ -337,6 +337,7 @@ export const createRpc = ({ common, chain, concurrency = 25, }) => {
                         params: ["newHeads"],
                         onData: async (data) => {
                             if (data.error === undefined && data.result !== undefined) {
+                                data.result.size = data.result.size ?? '0x0';
                                 onBlock(data.result);
                                 common.logger.debug({
                                     service: "rpc",
